applications/minotari_app_grpc/src/authentication/server_interceptor.rs

```diff
@@ -34,23 +35,41 @@ pub struct ServerAuthenticationInterceptor {
 }
 
 impl ServerAuthenticationInterceptor {
-    pub fn new(auth: GrpcAuthentication) -> Self {
-        Self { auth }
+    pub fn new(auth: GrpcAuthentication) -> Result<Self, String> {
+        // The server password needs to be hashed for later client verification
+        let processed_auth = match auth {
+            GrpcAuthentication::None => auth,
+            GrpcAuthentication::Basic { username, password } => GrpcAuthentication::Basic {
+                username,
+                password: create_salted_hashed_password(password.reveal())
+                    .map_err(|e| e.to_string())?
+                    .as_str()
+                    .into(),
+            },
+        };
+
+        Ok(Self { auth: processed_auth })
     }
 
     fn handle_basic_auth(
         &self,
         req: Request<()>,
         valid_username: &str,
-        valid_password: &[u8],
+        valid_phc_password: &SafePassword,
     ) -> Result<Request<()>, Status> {
         match req.metadata().get(AUTHORIZATION.as_str()) {
             Some(t) => {
-                let val = t.to_str().map_err(unauthenticated)?;
-                let credentials = BasicAuthCredentials::from_header(val).map_err(unauthenticated)?;
-                credentials
-                    .validate(valid_username, valid_password)
+                // Parse the provided header into (temporary) credentials for easier access
+                let header = t.to_str().map_err(unauthenticated)?;
+                let header_credentials = BasicAuthCredentials::from_header(header).map_err(unauthenticated)?;
```

Comment:
I'm considering refactoring to move header parsing out of `BasicAuthCredentials` altogether, since there should no longer be any reason to parse valid credentials in this way, and parsing the header credentials to this type is not idiomatic and only present out of necessity.